widgets/search-box.js

```diff
@@ -1,46 +1,59 @@
-var React = require('react');
-
 var utils = require('../lib/utils.js');
+var forEach = require('lodash/collection/forEach');
 
 function searchbox(params) {
-  var SearchBox = require('../components/SearchBox');
-
-  var container = utils.getContainerNode(params.container);
-  var isFocused = false;
+  // Hook on an existing input, or add one if none targeted
+  var input = utils.getContainerNode(params.container);
+  if (input.tagName !== 'INPUT') {
+    var newInput = document.createElement('input');
```

Comment:
same as:

```js
input = input.appendChild(document.createElement('input');
```

Because appendChild sends back the appended node.

Only for info, both ways are fine